sql/sys_vars.cc

```diff
@@ -278,6 +278,10 @@ static bool check_session_admin(sys_var *self MY_ATTRIBUTE((unused)), THD *thd,
   DBUG_ASSERT(self->scope() !=
               sys_var::GLOBAL);  // don't abuse check_session_admin()
   Security_context *sctx = thd->security_context();
+
+  /* Skip ACL checks for SET DEBUG and SET DEBUG_SYNC */
```

Comment:
This is not specific to only DEBUG and DEBUG_SYNC. Probably we should change it to.
```suggestion
  /* Skip ACL checks for SET commands */
```